lib/money/money.rb

```diff
@@ -340,14 +340,32 @@ def allocate_max_amounts(maximums)
   # @example
   #   Money.new(100, "USD").split(3) #=> [Money.new(34), Money.new(33), Money.new(33)]
   def split(num)
+    calculate_splits(num).sum { |value, count| Array.new(count, value) }
```

Comment:
Are you sure this works? You can sum array using `sum` without argument.

```
>> {1 => 2}.sum { |value, count| Array.new(count, value) }
Traceback (most recent call last):
        4: from /opt/rubies/2.5.3/bin/irb:11:in `<main>'
        3: from (irb):7
        2: from (irb):7:in `sum'
        1: from (irb):7:in `+'
TypeError (Array can't be coerced into Integer)
```

Reply:
Test were passing since in dev we are adding a dependency on Rails which adds ActiveSupport to add it. I've created this PR to fix this: https://github.com/Shopify/money/pull/180